src/Lykke.MatchingEngine.Connector/Services/ClientTcpSocket.cs

```diff
@@ -80,7 +80,7 @@ public void Start()
             if (_working)
                 throw new Exception("Client socket has already started");
             _working = true;
-            SocketThread();
+            SocketThread().Wait();
```

Comment:
It's better to use GetAwaiter().GetResult() instead of Wait()

Reply:
The change is incorrect. Before it worked as  "Fire and forget". SocketThread did not block the caller. Now it does.